flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/batch/table/TableSinkITCase.scala

```diff
@@ -207,4 +212,336 @@ class TableSinkITCase extends BatchTestBase {
+         |SELECT sum(y) FROM MyTable GROUP BY x
+         |""".stripMargin).await()
+    val expected = List(
+      "null,2021,1,0.1",
+      "null,2021,1,0.4",
+      "null,2021,1,1.0",
+      "null,2021,1,2.2",
+      "null,2021,1,3.9")
+    val result = TestValuesTableFactory.getResults("testSink")
+    assertEquals(expected.sorted, result.sorted)
+  }
+
+  @Test
+  def testFullInsertWithPartitionAndComputedColumn(): Unit = {
+    tEnv.executeSql(
+      s"""
+         |CREATE TABLE testSink (
+         |  `a` INT,
+         |  `b` AS `a` + 1,
+         |  `c` STRING,
+         |  `d` INT,
+         |  `e` DOUBLE
+         |)
+         |PARTITIONED BY (`c`, `d`)
+         |WITH (
+         |  'connector' = 'values',
+         |  'sink-insert-only' = 'true'
+         |)
+         |""".stripMargin)
+
+    registerCollection("MyTable", simpleData2, simpleType2, "x, y", nullableOfSimpleData2)
+
+    tEnv.executeSql(
+      s"""
+         |INSERT INTO testSink PARTITION(`c`='2021', `d`=1) (a, e)
+         |SELECT x, sum(y) FROM MyTable GROUP BY x
+         |""".stripMargin).await()
+    val expected = List(
+      "1,2021,1,0.1",
+      "2,2021,1,0.4",
+      "3,2021,1,1.0",
+      "4,2021,1,2.2",
+      "5,2021,1,3.9")
+    val result = TestValuesTableFactory.getResults("testSink")
+    assertEquals(expected.sorted, result.sorted)
+  }
+
+  @Test
+  def testPartialInsertWithDynamicPartitionAndComputedColumn1(): Unit = {
+    tEnv.executeSql(
+      s"""
+         |CREATE TABLE testSink (
+         |  `a` INT,
+         |  `b` AS `a` + 1,
+         |  `c` STRING,
+         |  `d` INT,
+         |  `e` DOUBLE
+         |)
+         |PARTITIONED BY (`c`, `d`)
+         |WITH (
+         |  'connector' = 'values',
+         |  'sink-insert-only' = 'true'
+         |)
+         |""".stripMargin)
+
+    registerCollection("MyTable", simpleData2, simpleType2, "x, y", nullableOfSimpleData2)
+
+    tEnv.executeSql(
+      s"""
+         |INSERT INTO testSink (e)
+         |SELECT sum(y) FROM MyTable GROUP BY x
+         |""".stripMargin).await()
+    val expected = List(
+      "null,null,null,0.1",
+      "null,null,null,0.4",
+      "null,null,null,1.0",
+      "null,null,null,2.2",
+      "null,null,null,3.9")
+    val result = TestValuesTableFactory.getResults("testSink")
+    assertEquals(expected.sorted, result.sorted)
+  }
+
+  @Test
+  def testPartialInsertWithDynamicPartitionAndComputedColumn2(): Unit = {
+    tEnv.executeSql(
+      s"""
+         |CREATE TABLE testSink (
+         |  `a` INT,
+         |  `b` AS `a` + 1,
+         |  `c` STRING,
+         |  `d` INT,
+         |  `e` DOUBLE
+         |)
+         |PARTITIONED BY (`c`, `d`)
+         |WITH (
+         |  'connector' = 'values',
+         |  'sink-insert-only' = 'true'
+         |)
+         |""".stripMargin)
+
+    registerCollection("MyTable", simpleData2, simpleType2, "x, y", nullableOfSimpleData2)
+
+    tEnv.executeSql(
+      s"""
+         |INSERT INTO testSink (c, d, e)
```

Comment:
Could you add a test which reorder the specified columns, e.g. `INSERT INTO .. (e, c, d)`. I tried in my local machine and it fails.

Reply:
fixed.